pytorch_lightning/plugins/training_type/ipu.py

```diff
@@ -109,14 +109,23 @@ def __init__(
                 options["autoReport.directory"] = self.autoreport_dir
             os.environ["POPLAR_ENGINE_OPTIONS"] = json.dumps(options)
 
-    def setup(self) -> None:
+    def setup(self, trainer: "pl.Trainer") -> None:
         # set the `accumulate_grad_batches` property as early as possible
         self._handle_gradient_accumulation_steps()
 
         # patch the dataloader creation function with the custom `poptorch.DataLoader`.
         # this violates the intended control flow for the plugins, but since this is experimental, we have chosen
         # to use the simpler solution before adding abstractions to override the `DataLoader` class
         self.lightning_module.trainer._update_dataloader = self._convert_to_poptorch_loader
+        super().setup(trainer)
+
+    def setup_optimizers(self, trainer: "pl.Trainer") -> None:
+        # refactor after move accelerator into strategy @four4fish
+        # RFC: I think set optimizer related logic should be in strategy instead of accelerator.
+        if len(self.optimizers) > 1:
```

Comment:
```suggestion
        # TODO (@four4fish): refactor after move accelerator into strategy
        if len(self.optimizers) > 1:
```
Let's do the discussion on the main #10416 about

> RFC: I think set optimizer related logic should be in strategy instead of accelerator.